inventory_management_system_api/cli/create.py

```diff
@@ -30,8 +48,137 @@ def system_type():
+        display_indexed_system_types(system_types)
+        _, dst_system_type = ask_user_for_index_selection(
+            "Please enter the index of the [green]dst_system_type[/]", system_types
+        )
+
+        # Output the existing usage statuses and obtain a user selected dst_usage_status
+        console.print("Below is a list of the current usage statuses available:")
+        display_indexed_usage_statuses(usage_statuses)
+        _, dst_usage_status = ask_user_for_index_selection(
+            "Please enter the index of the [green]dst_usage_status[/]", usage_statuses
+        )
+    elif rule_type == "moving":
+        # Output the existing types and obtain a user selected src_system_type and dst_system_type
+        console.print("Below is a list of the current system types available:")
+        display_indexed_system_types(system_types)
+        _, src_system_type = ask_user_for_index_selection(
+            "Please enter the index of the [green]src_system_type[/]", system_types
+        )
+        _, dst_system_type = ask_user_for_index_selection(
+            "Please enter the index of the [green]dst_system_type[/]", system_types
+        )
+
+        # Output the existing usage statuses and obtain a user selected dst_usage_status
+        console.print("Below is a list of the current usage statuses available:")
+        display_indexed_usage_statuses(usage_statuses)
+        _, dst_usage_status = ask_user_for_index_selection(
+            "Please enter the index of the [green]dst_usage_status[/]", usage_statuses
+        )
+    elif rule_type == "deletion":
+        # Output the existing types and obtain a user selected src_system_type
+        console.print("Below is a list of the current system types available:")
+        display_indexed_system_types(system_types)
+        _, src_system_type = ask_user_for_index_selection(
+            "Please enter the index of the [green]src_system_type[/]", system_types
+        )
+
+    return src_system_type, dst_system_type, dst_usage_status
+
+
+def display_user_constructed_rule(
+    rule_type: RuleType, src_system_type: SystemTypeOut, dst_system_type: SystemTypeOut, dst_usage_status: SystemTypeOut
+):
+    """Displays a user constructed rule obtained from `get_user_constructed_rule`."""
+
+    # Output the selected rule
+    console.print(f"You are creating the following {rule_type} rule:")
+    table = Table("src_system_type_id", "dst_system_type_id", "dst_usage_status_id")
+    table.add_row(
+        (f"{src_system_type.id} [orange1]({src_system_type.value})[/]" if src_system_type else "None"),
+        (f"{dst_system_type.id} [orange1]({dst_system_type.value})[/]" if dst_system_type else "None"),
+        (f"{dst_usage_status.id} [orange1]({dst_usage_status.value})[/]" if dst_usage_status else "None"),
+    )
+    console.print(table)
+    console.print()
+
+    # Output an explanation of the rule
+    display_rule_explanation(src_system_type, dst_system_type, dst_usage_status)
```

Comment:
I get the following warning for `dst_usage_status`: `Expected type 'UsageStatusOut | None', got 'SystemTypeOut' instead `